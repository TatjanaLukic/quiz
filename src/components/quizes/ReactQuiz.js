import React, { useState } from "react";
import Result from "../Result";
import QuestionList from "../QuestionList";
import Title from "../Title";
import {reactQuestions} from '../../data/reactQuestions';

const ReactQuiz = () => {
  const [score, setScore] = useState(0);
  const [showResult, setShowResult] = useState(false);
  const [loading, setLoading] = useState(false);

  const setCalculatedScore = calculatedScore => {
    setScore(calculatedScore);
    setShowResult(true);
  };

  return (
    <div style={{ background: "#8f34eb" }}>
      <Title title="React Quiz" />
      <div className="ui container">
        {showResult && <Result score={score} />}
        <QuestionList setCalculatedScore={setCalculatedScore} questions={reactQuestions} loading = {loading} />
      </div>
    </div>
  );
};

export default ReactQuiz;
