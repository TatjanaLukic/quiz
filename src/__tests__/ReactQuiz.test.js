import React from 'react';

import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ReactQuiz from '../components/quizes/ReactQuiz';

configure({adapter: new Adapter()});

describe('<ReactQuiz/>',() => {

it("spanshot matches", () => {
    const tree = shallow(<ReactQuiz />);
    expect(toJson(tree)).toMatchSnapshot();
  });

});