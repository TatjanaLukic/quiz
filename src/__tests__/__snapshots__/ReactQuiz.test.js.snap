// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactQuiz/> spanshot matches 1`] = `
<div
  style={
    Object {
      "background": "#8f34eb",
    }
  }
>
  <Title
    title="React Quiz"
  />
  <div
    className="ui container"
  >
    <QuestionList
      loading={false}
      questions={
        Array [
          Object {
            "A": "Library for UI.",
            "B": "Complete framework for front-end apps.",
            "answer": "A",
            "category": "basic",
            "id": 1,
            "q": "What is React?",
          },
          Object {
            "A": "Yes.",
            "B": "No.",
            "answer": "A",
            "category": "basic",
            "id": 2,
            "q": "React creates Virtual DOM?",
          },
          Object {
            "A": "Data and functions.",
            "B": "Just data.",
            "answer": "A",
            "category": "basic",
            "id": 3,
            "q": "Props can be:",
          },
          Object {
            "A": "State of component is passed by parent component.",
            "B": "State of component is defined and owned by component itself.",
            "answer": "B",
            "category": "basic",
            "id": 4,
            "q": "What is correct?",
          },
          Object {
            "A": "Asynchronous function.",
            "B": "Synchronous function.",
            "answer": "A",
            "category": "basic",
            "id": 5,
            "q": "setState is?",
          },
        ]
      }
      setCalculatedScore={[Function]}
    />
  </div>
</div>
`;
